packages/task/src/browser/task-service.ts

```diff
@@ -700,12 +700,15 @@ export class TaskService implements TaskConfigurationClient {
             const taskName = this.taskNameResolver.resolve(task);
             const terminalId = matchedRunningTaskInfo.terminalId;
             if (terminalId) {
-                const terminal = this.terminalService.getById(this.getTerminalWidgetId(terminalId));
-                if (terminal && task.presentation) {
-                    if (task.presentation.focus) { // assign focus to the terminal if presentation.focus is true
-                        this.terminalService.open(terminal, { mode: 'activate' });
-                    } else if (task.presentation.reveal === RevealKind.Always) { // show the terminal but not assign focus
-                        this.terminalService.open(terminal, { mode: 'reveal' });
+                const widgetId = this.getTerminalWidgetId(terminalId);
+                if (widgetId) {
+                    const terminal = this.terminalService.getById(widgetId);
```

Comment:
could you clarify for me why `getById()` is used here.
I mean you use:
- `getByTerminalId` when `onTaskExit`
- `getByTerminalId` when `onOutputProcessed`

but 
- `this.getTerminalWidgetId(terminalId)` and then `getById()` when task is running

I see you have `terminalId` at this step, but another way is used to get a corresponding terminal.

thanks in advance!

Reply:
moreover, `getTerminalWidgetId()` uses `getByTerminalId()` inside.

`this.getTerminalWidgetId(terminalId)` ->
`this.terminalService.getByTerminalId(terminalId)`->  you have terminal at this step ->
`terminalWidget.id` -> 
`this.terminalService.getById(widgetId)`

Looks strange for me...

Reply:
Yes you are right. With the new function `terminalService.getByTerminalId()` added I can get the terminal instance directly from all places.